code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,278 @@
+	/// destination tagging for the mail sorter
+	var/sortTag = 0
+	/// who this mail is for and who can open it
+	var/mob/recipient
+	/// how many goodies this mail contains
+	var/goodie_count = 1
+	/// Goodies which can be given to anyone. The base weight for cash is 56. For there to be a 50/50 chance of getting a department item, they need 56 weight as well.
+	var/list/generic_goodies = list(
+		/obj/item/stack/spacecash/c50 = 10,
+		/obj/item/stack/spacecash/c100 = 25,
+		/obj/item/stack/spacecash/c200 = 15,
+		/obj/item/stack/spacecash/c500 = 5,
+		/obj/item/stack/spacecash/c1000 = 1,
+	)
+	// Overlays (pure fluff)
+	/// Does the letter have the postmark overlay?
+	var/postmarked = TRUE
+	/// Does the letter have a stamp overlay?
+	var/stamped = TRUE
+	/// List of all stamp overlays on the letter.
+	var/list/stamps = list()
+	/// Maximum number of stamps on the letter.
+	var/stamp_max = 1
+	/// Physical offset of stamps on the object. X direction.
+	var/stamp_offset_x = 0
+	/// Physical offset of stamps on the object. Y direction.
+	var/stamp_offset_y = 2
+
+	///mail will have the color of the department the recipient is in.
+	var/static/list/department_colors
+
+/obj/item/mail/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	goodie_count = 2
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/Initialize()
+	. = ..()
+	AddElement(/datum/element/item_scaling, 0.5, 1)
+	if(!department_colors?.len)
+		department_colors = list(
+		ACCOUNT_CIV = COLOR_WHITE,
+		ACCOUNT_ENG = COLOR_PALE_ORANGE,
+		ACCOUNT_SCI = COLOR_PALE_PURPLE_GRAY,
+		ACCOUNT_MED = COLOR_PALE_BLUE_GRAY,
+		ACCOUNT_SRV = COLOR_PALE_GREEN_GRAY,
+		ACCOUNT_CAR = COLOR_BEIGE,
+		ACCOUNT_SEC = COLOR_PALE_RED_GRAY,
+		)
+
+	// Icons
+	// Add some random stamps.
+	if(stamped == TRUE)
+		var/N = rand(1, stamp_max)
+		for(var/i = 1, i <= N, i++)
+			var/X = rand(2, 6)
+			stamps += list("stamp_[X]")
+	update_icon()
+
+/obj/item/mail/update_icon()
+	. = ..()
+	cut_overlays()
+
+	var/bonus_stamp_offset = 0
+	for(var/S in stamps)
+		var/image/SI = image(
+			icon = icon,
+			icon_state = S,
+			pixel_x = stamp_offset_x,
+			pixel_y = stamp_offset_y + bonus_stamp_offset
+		)
+		// Stops postmarks from inheriting letter color.
+		// http://www.byond.com/docs/ref/#/atom/var/appearance_flags
+		SI.appearance_flags |= RESET_COLOR
+		add_overlay(SI)
+		bonus_stamp_offset -= 5
+
+	if(postmarked == TRUE)
+		var/image/PMI = image(
+			icon = icon,
+			icon_state = "postmark",
+			pixel_x = stamp_offset_x + rand(-3, 1),
+			pixel_y = stamp_offset_y + rand(bonus_stamp_offset + 3, 1)
+		)
+		PMI.appearance_flags |= RESET_COLOR
+		add_overlay(PMI)
+
+/obj/item/mail/attackby(obj/item/W, mob/user, params)
+	// Destination tagging
+	if(istype(W, /obj/item/dest_tagger))
+		var/obj/item/dest_tagger/O = W
+
+		if(sortTag != O.currTag)
+			var/tag = uppertext(GLOB.TAGGERLOCATIONS[O.currTag])
+			to_chat(user, "<span class='notice'>*[tag]*</span>")
+			sortTag = O.currTag
```

Comment:
Is this not copied and pasted code from packaged objects?

This should probably be an element or similar.